clients/orderbook/src/worker.rs

```diff
@@ -165,10 +169,45 @@ where
 			pending_withdrawals: vec![],
 			last_finalized_block: 0,
 			sync_state_map: Default::default(),
+			last_block_snapshot_generated: 0,
+			latest_stid: 0,
 		}
 	}
 
-	pub fn process_withdraw(&mut self, withdraw: WithdrawalRequest) -> Result<(), Error> {
+	/// The function checks whether a snapshot of the blockchain should be generated based on the
+	/// pending withdrawals and block interval.
+	///
+	/// # Parameters
+	/// - &self: a reference to an instance of a struct implementing some trait
+	/// # Returns
+	/// - bool: a boolean indicating whether a snapshot should be generated
+	pub fn should_generate_snapshot(&self) -> bool {
+		// Get the snapshot generation intervals from the runtime API for the last finalized block
```

Comment:
Ideally, this must be done in pallet but I don't see any reason why this will fail too. Do you guys find any corner cases here? @felixfaisal @nuel77